lib/tasks/interfaces.rake

```diff
@@ -0,0 +1,39 @@
+namespace :interfaces do
+  desc <<-END_DESC
+Removes old interfaces that match ignored interfaces pattern setting.
+
+This is useful when you change the ignored pattern setting.
+
+Examples:
+  # foreman-rake interfaces:clean
+END_DESC
+
+  task :clean => :environment do
+    puts 'Starting ingnored interfaces clean up...'
+    cleaner = InterfaceCleaner.new.clean!
+    puts "Finished, cleaned #{cleaner.deleted_count} interfaces"
+    unless cleaner.primary_hosts.empty?
+      puts "Following hosts have ignored interface set as primary, please set other interface as primary and rerun the task:"
+      print_host_queries(cleaner.primary_hosts)
+    end
+    unless cleaner.provision_hosts.empty?
+      puts "Following hosts have ignored interface set as provision, please set other interface as provisioning and rerun the task:"
+      print_host_queries(cleaner.provision_hosts)
+    end
+  end
+
+  def print_host_queries(hosts)
+    Host.unscoped.where(id: hosts.uniq).pluck(:name).in_groups_of(hosts_group_count, false) do |names|
+      query = "id ^ (#{names.join(',')})"
```

Comment:
Here is one thing - you join names but query ids, which does not work: `http://box.home.lan/hosts?search=id+%5E+%28carl-magpusao.home.lan%29` does not find it. You need to pluck :id here.